lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Utils.pm

```diff
@@ -196,6 +197,30 @@ sub serialize_annotation {
     return;
 }
 
+sub serialize_artist_credit {
+    my ($into, $entity, $inc, $stash, $toplevel) = @_;
+
+    # In contrast to most other serialize_* methods,
+    # this method has to be called only when appropriate.
+    # Because the conditions to decided whether or not to serialize
+    # artist credit vary a lot depending on the entity type,
+    # no further check is made here.
+
+    my $artist_credit = $entity->artist_credit;
+
+    $into->{'artist-credit'} = [map {
+        {
+            'name' => $_->name,
+            'joinphrase' => $_->join_phrase,
+            'artist' => serialize_entity($_->artist, $inc, $stash),
+        }
+    } @{ $artist_credit->{names} }];
+
+    $into->{'artist-credit-id'} = $artist_credit->{gid};
```

Comment:
This seems good for now - should we have a ticket to change AC to an object whenever we do breaking changes for ws/2.5 or ws/3 or whatever we end up calling it?

Reply:
It is covered by MBS-9829 already.